tests/integration/components/sl-textarea-test.js

```diff
@@ -0,0 +1,305 @@
+        'SMALL',
+        "Label's text-danger <small> is rendered"
+    );
+});
+
+test(
+    'If "label" property is not populated, "optional" and "required" elements are not rendered even if populated',
+    function( assert ) {
+        this.render( hbs`
+            {{#sl-textarea optional=true required=true}}
+            {{/sl-textarea}}
+        ` );
+
+        assert.strictEqual(
+            this.$( '>:first-child' ).find( 'label > .text-info' ).length,
+            0,
+            "Label's text-info is not rendered"
+        );
+
+        assert.strictEqual(
+            this.$( '>:first-child' ).find( 'label > .text-danger' ).length,
+            0,
+            "Label's text-danger is not rendered"
+        );
+    }
+);
+
+test( 'If "label" property is not populated, label element is not rendered', function( assert ) {
+    this.render( hbs`
+        {{#sl-textarea label=""}}
+        {{/sl-textarea}}
+    ` );
+    assert.strictEqual(
+        Ember.typeOf( this.$( '>:first-child' ).find( 'label' ).prop( 'for' ) ),
+        'undefined',
+        'Label element is not rendered'
+    );
+});
+
+test( 'If "label" property is populated, label element is rendered', function( assert ) {
+    this.set( 'label', 'test' );
+    this.render( hbs`
+        {{#sl-textarea label="test"}}
+        {{/sl-textarea}}
+    ` );
+
+    const label = this.$(
+        'label[for="' + this.$( '>:first-child' ).find( 'textarea' ).prop( 'id' ) + '"]'
+    );
+
+    assert.strictEqual(
+        label.length,
+        1,
+        'Label is present'
+    );
+
+    assert.strictEqual(
+        Ember.$.trim( label.text() ),
+        this.get( 'label' ),
+        'Label text is expected value'
+    );
+});
+
+test( 'If "label" property is populated, "for" attribute is expected value', function( assert ) {
+    this.set( 'label', 'Test Label' );
+    this.render( hbs`
+        {{#sl-textarea label=label}}
+        {{/sl-textarea}}
+    ` );
+
+    assert.strictEqual(
+        this.$( '>:first-child' ).find( 'label' ).prop( 'for' ),
+        this.$( '>:first-child' ).find( 'textarea' ).prop( 'id' ),
+        'Label "for" property matches textarea\'s "id" property'
+    );
+});
+
+test( '"selectionStart" is supported', function( assert ) {
+    this.render( hbs`
+        {{#sl-textarea selectionStart=2 value="testValue"}}
+        {{/sl-textarea}}
+    ` );
+    this.$( '>:first-child' ).find( 'textarea' ).get( 0 ).setSelectionRange( 2, 8 );
+
+    assert.strictEqual(
+        this.$( '>:first-child' ).find( 'textarea' ).prop( 'selectionStart' ),
+        2,
+        'Textarea "selectionStart" property is expected value'
+    );
+});
+
+test( '"selectionEnd" is supported', function( assert ) {
+    this.render( hbs`
+        {{#sl-textarea selectionEnd=8 value="testValue"}}
+        {{/sl-textarea}}
+    ` );
+
+    this.$( '>:first-child' ).find( 'textarea' ).get( 0 ).setSelectionRange( 2, 8 );
```

Comment:
I see a few issues. `selectionStart` and `selectionEnd` are properties and not HTML attributes that can get set on the element. In the [component code] (https://github.com/softlayer/sl-ember-components/blob/c79f24b99c8850eba970938c669f592af1e4ad1b/addon/components/sl-textarea.js#L93) and documentation we refer to them as attributes that will get set on the textarea. This is not accurate as far as my experimentation goes, [this works] (http://jsfiddle.net/erbs6v8j/) [this] (http://jsfiddle.net/8oyw5wph/4/) doesn't. And even if we were able to set these properties as attributes on the element, we are not currently doing anything with the properties, just because they are passed to the component, the component doesn't do anything with them, they are defined but never used or set on the element in any way.

This test passes because we are testing that the browser returns the correct value for selected text.